File: v2/dtplan_api/src/main/java/com/dtplan/domain/execucaoExercicio/dto/ExecucaoExercicioDTO.java
package com.dtplan.domain.execucaoExercicio.dto;

import com.dtplan.domain.execucaoExercicio.ExecucaoExercicio;

public record ExecucaoExercicioDTO(
        Long id,
        Integer seriesRealizadas,
        Integer repeticoesRealizadas,
        Double peso,
        String observacao,
        Long exercicioId,
        String exercicioNome
) {
    public ExecucaoExercicioDTO(ExecucaoExercicio execucaoExercicio) {
        this(
                execucaoExercicio.getId(),
                execucaoExercicio.getSeriesRealizadas(),
                execucaoExercicio.getRepeticoesRealizadas(),
                execucaoExercicio.getPeso(),
                execucaoExercicio.getObservacao(),
                execucaoExercicio.getExercicio() != null ? execucaoExercicio.getExercicio().getId() : null,
                execucaoExercicio.getExercicio() != null ? execucaoExercicio.getExercicio().getNome() : "Exercício não especificado"
        );
    }
}

File: v2/dtplan_web/src/app/core/models/execucaoExercicio.dto.ts
export interface ExecucaoExercicioDTO {
  id: number;
  seriesRealizadas: number;
  repeticoes: number;
  carga: number;
  exercicioId: number;
  exercicioNome: string;
}

export interface ListarExecucaoExercicioDTO {
  id: number;
  seriesPrevistas: number;
  seriesRealizadas: number;
  repeticoesPrevistas: number;
  repeticoesRealizadas: number;
  carga: number;
  exercicioId: number;
  exercicioNome: string;
}